include/soci/soci-backend.h

```diff
@@ -105,11 +112,16 @@ inline db_type to_db_type(data_type dt)
         case dt_unsigned_long_long: return db_uint64;
         case dt_blob:               return db_blob;
         case dt_xml:                return db_xml;
+        default:
+            throw soci_error("unsupported data_type");
```

Comment:
I'd add `case dt_wstring: break;` here and move this `throw` below, after the `switch`. This has the following advantages:

1. We'll get a warning from `gcc -Wswitch` if/when new elements are added to `data_type` enum.
2. We can get rid of MSVC warning without using pragmas.

Reply:
Okay, I made the change. But keep in mind that this required adding `dt_wstring` to the deprecated `data_type` enum, which you didn't like before.